### PR TITLE
Module changes for auto ami update

### PR DESCRIPTION
**Pull request by sobhanagit**
To implement support for `ami_auto_update` object inside `parameters` for `sheduled_tasks`

# Jira Ticket

https://spotinst.atlassian.net/browse/SPOTAUT-17719

### CODE REVIEW

File: examples/basic/main.tf [hunk 1]
```diff
@@ -3,13 +3,39 @@ module "ocean-aws-k8s" {
...
-  security_groups             = ["sg-123456789","sg-123456789"]
-
+  cluster_name                   = "EKS-Example"
+  controller_id                  = "EKS controller id"
+  region                         = "us-west-2"
+  subnet_ids                     = ["subnet-12345678","subnet-12345678"]
+  min_size                       = 0
+  worker_instance_profile_arn    = "arn:aws:iam::253244684816:instance-profile/eks-4cc62bd6-5b16-e9be-84f2-8ae3837be2f5"
+  security_groups                = ["sg-123456789","sg-123456789"]
+  ami_id                         = "ami-02620bd547be274fb"
```
Comment (chandra1-n) [inline examples/basic/main.tf:13]: please remove the actual value

File: examples/basic/main.tf [hunk 2]
```diff
@@ -3,13 +3,39 @@ module "ocean-aws-k8s" {
...
+  shutdown_hours                 = {
+  is_enabled                     = true
```
Comment (chandra1-n) [inline examples/basic/main.tf:15]: change the value to false

File: examples/basic/main.tf [hunk 3]
```diff
@@ -3,13 +3,39 @@ module "ocean-aws-k8s" {
...
+  time_windows                   = ["Sat:08:00-Sun:08:00"]
+  }
+  tasks                          = [{
+    is_enabled                   = false
+    cron_expression              = "0 1 * * *"
+    task_type                    = "amiAutoUpdate"}]
+  ami_auto_update                = {
+    apply_roll                   = true
```
Comment (chandra1-n) [inline examples/basic/main.tf:23]: change the value to false

File: variables.tf [hunk 4]
```diff
@@ -333,6 +334,39 @@ variable "tasks" {
...
+    respect_pdb = optional(bool,null)
+  })
+  default     = null
+  description = "amiAutoUpdate cluster roll object"
+}
+
+## parameter cluster roll ##
+variable "parameters_cluster_roll" {
+  type = object({
+    batch_min_healthy_percentage = optional(number,50)
```
Comment (chandra1-n) [inline variables.tf:362]: test keeping it as null instead of 50

File: variables.tf [hunk 5]
```diff
@@ -333,6 +334,39 @@ variable "tasks" {
...
+    minor_version = optional(bool,false)
+    patch         = optional(bool,false)
+  })
+  default     = null
+  description = "amiAutoUpdate cluster roll object"
+}
+## amiAutoUpdate cluster roll ##
+variable "ami_auto_update_cluster_roll" {
+  type = object({
+    batch_min_healthy_percentage = optional(number,50)
```
Comment (chandra1-n) [inline variables.tf:350]: test keeping it as null instead of 50

**Review by chandra1-n — CHANGES_REQUESTED** [review]
please fix the review comments

File: variables.tf [hunk 6]
```diff
@@ -303,7 +303,8 @@ variable "conditioned_roll" {
 variable "conditioned_roll_params" {
   type        = list(string)
   default     = null
-  description = "Customized List of conditioned roll params, it is valid only when conditioned_roll set to true"}
+  description = "Customized List of conditioned roll params, it is valid only when conditioned_roll set to true"
```
Comment (chandra1-n) [inline variables.tf:306]: this fix is merged today, please take latest

**Review by anuragsharma-123 — APPROVED** [review]

**Review by chandra1-n — APPROVED** [review]

**Review by anuragsharma-123 — APPROVED** [review]

**Pull request merged**